### PR TITLE
Add v7v8 simple upgrade test via starship,

### PR DESCRIPTION
**Pull request by Anmol1696**
## 1. Overview
Using Starship to run sample v7v8 upgrade tests.

## 2. Implementation details
-  add ci for running e2e tests that connects to digital ocean kubernetes cluster
- rename shuttle to starship
- add handy scripts for starship for portforwarding

## 3. How to test/use
Pre: Connect to a kubernetes cluster, or start one locally with `kind`
- Connect to k8s:
  ```
  kubectl get pods
  ```
- Infra setup: 
  ```bash
  cd starship/v7v8/
  make start
  # OR breakdown the start command
  make setup-helm   # fetches helm charts required
  make install      # runs the helm on the k8s cluster to bring up the infra
  make port-forward # port-forwards required ports to localhost
  ```
- Tests:
  ```bash
  cd starship/v7v8/
  make test
  ```

## 4. Checklist

- [x] Does the Readme need to be updated?

## 5. Limitations (optional)
- For every chain upgrade, would require a clone of the `starship/v7v8/` dir and additional job in the repo
- When to run the tests on the CI is still to be figured out (currently running the tests on PR and Push to main if changes to starship dir)

## 6. Future Work (optional)
- Add more test cases for the specific chain upgrade with the tests.

### CODE REVIEW

**Comment by xlab** [comment]
@Anmol1696 few questions, the code is still to review, I am testing this locally.

1) can you provide and example of config that doesn't do upgrade? Just starts the chain of specified version that can be used for tests

2) I see that the image is being pulled `anmol1696/persistencecore:latest`, is there a way to specify in config or build in-place when Dockerfile exists?

Screenshot 2023-05-11 at 22 12 08

relevant to previous question, as ideally wanted to specify exact commit / tag to use.

3) if I don't specify relayer in the config, a **hermes** pod is still being initialized
Screenshot 2023-05-11 at 22 13 44

**Comment by Anmol1696** [comment]
@xlab here is the issue on Starship I created for your request. I think this makes sense, let me add it to Starship and get back to you. 

Just FYI you can have a look at current alternatives too to have an idea: https://github.com/cosmology-tech/starship/issues/90

**Comment by xlab** [comment]
@Anmol1696 will try with custom docker, thanks!

**Comment by Anmol1696** [comment]
@xlab If you upgrade the helm chart to version `0.1.24` you should be able to use the new build system as well, for inplace building of chain via source branch or tag.

Please try it out and let me know.

File: .github/workflows/e2e-upgrade.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          namespace: ${{ steps.set-namespace.outputs.namespace }}
+
+      - name: Run Tests
+        run: |
+          cd starship/v7v8/ 
+          go mod tidy
+          make test
+
+      # Starship resource cleanup on cluster
+      # todo(@anmol1696): change this to be post step of the action
```
Comment (xlab) [inline .github/workflows/e2e-upgrade.yaml:64]: Is this still in WIP?

**Review by xlab — APPROVED** [review]

**Comment by xlab** [comment]
@Anmol1696 check this please
https://github.com/persistenceOne/persistenceCore/actions/runs/4991290006/jobs/8937577778?pr=198#step:5:367
Screenshot 2023-05-16 at 14 06 11

**Comment by Anmol1696** [comment]
@xlab it was a transient error maybe cuz of not enough resources on the kubernetes cluster. Seems to be resolved now.
I would recommed @snj33v we have some sort of auto scaling on the nodes as well, or monitor the usage.

File: .github/workflows/e2e-upgrade.yaml (hunk 1 above)
Comment (Anmol1696) [reply .github/workflows/e2e-upgrade.yaml:64]: It is a todo, currently this step will work as we need it to, i just dont think it is super clean.

Once I have a new version of the starship-action, i will come back and fix this as well, hence the todo here.

This should not be blocking though.

**Pull request merged**